walk/walk.go

```diff
@@ -122,25 +124,26 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 		log.Printf("error loading .bazelignore: %v", err)
 	}
 
-	visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, c.RepoRoot, "", false)
-}
-
-func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnoredFunc, knownDirectives map[string]bool, updateRels *UpdateFilter, wf WalkFunc, dir, rel string, updateParent bool) {
-	if isBazelIgnored(rel) {
-		return
+	trie, err := buildTrie(c, isBazelIgnored)
+	if err != nil {
+		log.Fatalf("error walking the file system: %v\n", err)
 	}
 
+	visit(c, cexts, knownDirectives, updateRels, trie, wf, c.RepoRoot, "", false)
```

Comment:
Sorry I didn't see this and then this was merged. We can probably do this fairly easily.

Lets just make sure we do this (disable it by default) instead of reverting if any issues come up now that there has been a release.